worker/execute/session.py

```diff
@@ -0,0 +1,20 @@
+from util.network import get_local_ip, get_random_port
+
+
+class Session:
+    """
+    Base class for alternative types of job sessions.
+    """
+
+    protocol: str
+    ip: str = get_local_ip()
+    port: int
+
+    def __init__(self, protocol: str = "ws"):
+        self.protocol = protocol
+        self.ip = Session.ip
+        self.port = get_random_port()
```

Comment:
This seems like it would be a race condition waiting to happen. Should this not be moved to `process.py` where the port is opened and held open? Maybe it doesn't matter if it's just for testing.

Perhaps after `executa` starts it could somehow print out its listening port in a way that python is able to parse?